docs/source/api/protocol/index.rst

```diff
@@ -0,0 +1,27 @@
+`protocol`
+===========
+
+In `tf-encrypted`, a protocol represents a certain type of cryptographic protocol
+to achieve security.
+
+The goal is that you can easily play with or use different cryptographic methods
```

Comment:
We should probably document why you'd want to change protocols? e.g. for benchmarking purposes etc

cc @mortendahl